src/components/CodeAutocompleteField/CodeAutocompleteField.tsx

```diff
@@ -2,21 +2,39 @@ import React from "react";
 
 import { CLASSPREFIX as eccgui } from "../../configuration/constants";
 import AutoSuggestion, { AutoSuggestionProps } from "../AutoSuggestion/AutoSuggestion";
+import Spinner from "../Spinner/Spinner";
 
-export type CodeAutocompleteFieldProps = AutoSuggestionProps;
+export interface CodeAutocompleteFieldProps extends AutoSuggestionProps {
+    /** If the component should re-init when the initial value changes. Used when a value should be set externally.*/
+    reInitOnInitialValueChange?: boolean
+}
 
 /**
  * Input component that allows partial, fine-grained auto-completion, i.e. of sub-strings of the input string.
  * This is comparable to a one line code editor.
  *
  * Example usage: input of a path string offering auto-completion for each single part of the path.
  */
-export function CodeAutocompleteField({ className, ...otherProps }: CodeAutocompleteFieldProps) {
+export function CodeAutocompleteField({ className, reInitOnInitialValueChange = false, ...otherProps }: CodeAutocompleteFieldProps) {
+    const [reInit, setReInit] = React.useState<boolean>(false)
+
+    React.useEffect(() => {
```

Comment:
There was existing code related to this issue ( an issue from Manual testing) that I initially thought was redundant in the `AutoSuggestion.tsx`.  Adding it back with a minor change, seems like a better alternative to this. 

The previous code was like this


```
React.useEffect(() => {
        if (initialValue != null && cm) {
            dispatch({
                changes: { from: 0, to: cm?.state?.doc.length, insert: initialValue },
            });
        }
    }, [initialValue, cm]);
```

    
    
Taking away the cm dependency from there, seems to work and would be lot less code. Also,  that way we don't need props added anywhere else or even have to modify `CodeAutocompleteField`

Reply:
I will integrate this. The switch is still useful in some places, so I will probaby keep it.